dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java

```diff
@@ -0,0 +1,167 @@
+
+    static void newStackContext(final AppView appView) {
+        threadLocal.set(new StackContext(appView));
+    }
+
+    private static StackContext currentStackContext() {
+        return threadLocal.get();
+    }
+
+    static void currentSite(final String siteId) {
+        final StackContext currentStack = currentStackContext();
+        if(null == currentStack){
+            throw new IllegalStateException("Create stack hasn't been called yet.");
+        }
+        currentStack.currentSite = siteId;
+    }
+
+    static void pushSecret(final Map<String, Object> map) {
+        final StackContext context = currentStackContext();
+        if(null == context){
+            throw new IllegalStateException("Create stack hasn't been called yet.");
+        }
+        if (null == context.currentSite) {
+            throw new IllegalStateException(
+                    "Site must be pushed prior to pushing a secret.");
+        }
+        context.secretsBySite.computeIfAbsent(context.currentSite, k -> new ArrayList<>()).add(map);
+    }
+
+    static void disposeStackContext() {
+        threadLocal.remove();
+    }
+
+    static class StackContext {
+
+        final Map<String, List<Map<String,Object>>> secretsBySite = new HashMap<>();
+
+        final AppView appView;
+
+        String currentSite;
+
+        StackContext(final AppView appView) {
+           this.appView = appView;
+        }
+
+    }
+
+    public static class SecretHolder {
+
+        private final String name;
+        private final String label;
+        private final String hint;
+        private final Object value;
+
+        SecretHolder(final String name, final String label, final String hint, Object value) {
+            this.name = name;
+            this.value = value;
+            this.label = label;
+            this.hint = hint;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public Object getValue() {
+            return value;
+        }
+
+        public String getLabel() {
+            return label;
+        }
+
+        public String getHint() {
+            return hint;
+        }
+
+        @Override
+        public String toString() {
+            return "SecretHolder{" +
+                    "name='" + name + '\'' +
+                    ", value='" + value + '\'' +
+                    ", label='" + label + '\'' +
+                    ", hint='" + hint + '\'' +
+                    '}';
+        }
+    }
+
+
+    static String interpolateValues(final String inputJson) {
+        final StackContext context = ViewUtil.currentStackContext();
+        if (null == context) {
+            throw new IllegalStateException(
+                    "This method can only be called once a context has been previously created.");
+        }
+        try {
+            final RuntimeServices runtimeServices = RuntimeSingleton.getRuntimeServices();
+            final StringReader stringReader = new StringReader(inputJson);
```

Comment:
Codacy found an issue: [Ensure that resources like this StringReader object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)